io/io/test/CMakeLists.txt

```
@@ -9,3 +9,6 @@ ROOT_ADD_GTEST(TFile TFileTests.cxx LIBRARIES RIO)
 ROOT_ADD_GTEST(TBufferMerger TBufferMerger.cxx LIBRARIES RIO Imt Tree)
 ROOT_ADD_GTEST(TFileMerger TFileMergerTests.cxx LIBRARIES RIO Tree)
 ROOT_ADD_GTEST(TROMemFile TROMemFileTests.cxx LIBRARIES RIO Tree)
+if(uring AND NOT DEFINED ENV{ROOTTEST_IGNORE_URING})
+  ROOT_ADD_GTEST(RIoUring RIoUring.cxx LIBRARIES RIO uring)
```

Isn't `uring` inherited through `RIO`? I'd hope that you don't need to spell out `LIBRARIES uring` here:
```suggestion
  ROOT_ADD_GTEST(RIoUring RIoUring.cxx LIBRARIES RIO)
```

---

The missing piece here is that I didn't make `target_link_libraries` `PUBLIC` instead of `PRIVATE`. I think it is fixed in 23d2a9f4d0bba8f4e87541d0baaf8e9e930aa3c1